include/psa/crypto_struct.h

```diff
@@ -262,11 +262,33 @@ typedef struct psa_tls12_prf_key_derivation_s
 } psa_tls12_prf_key_derivation_t;
 #endif /* MBEDTLS_MD_C */
 
+typedef struct psa_key_derivation_input_buffer_s
+{
+    const uint8_t *data;
+    size_t length;
+    psa_key_derivation_step_t step;
+} psa_key_derivation_input_buffer_t;
+
+typedef struct psa_opaque_key_derivation_s
+{
+    psa_key_handle_t key;
+    uint8_t *salt;
+    size_t salt_length;
+    uint8_t *label;
```

Comment:
The label entry may be a bit more complex when CMAC support (e.g. KDF using counter mode) is added to this driver, later on.

Specifically there is an operation to append `context,` which is differentiated from the `label` given as input to the PRF (according to NIST Special Publication 800-108): `<iteration> || label || 0x00 || context || <length>`

Is the assumption that `label` would type in the structure would handle a concatenation of `label || 0x00 || context` or is there room to add `context` and `context_length`in this structure type?

I'm asking this question since the type here isn't HKDF or tls12_prf-specific...